### ISSUE TITLE
Add instructions to support dotenv?

### ISSUE THREAD

**Issue by jeromedalbert** [rails/webpacker#586]
I believe dotenv is fairly popular in the Ruby community.

Here's what I did to support a "Ruby-like" dotenv with Webpack:

```
yarn add dotenv
```

```javascript
// config/webpack/dotenv.js

const dotenv = require('dotenv');

const dotenvFiles = [
  `.env.${process.env.NODE_ENV}.local`,
  '.env.local',
  `.env.${process.env.NODE_ENV}`,
  '.env'
];

function loadEnv() {
  dotenvFiles.forEach((dotenvFile) => {
    dotenv.config({ path: dotenvFile, silent: true });
  });
}

module.exports = { loadEnv };
```

```javascript
// config/webpack/shared.js

...
const dotenv = require('./dotenv');

dotenv.loadEnv();

module.exports = {

  ...

  plugins: [
    new webpack.EnvironmentPlugin(process.env),
    ...
```

And here's how to use env variables in your JS code:

```bash
# .env

STRIPE_PUBLISHABLE_KEY=...
```

```javascript
// Example JS file

Stripe.setPublishableKey(process.env.STRIPE_PUBLISHABLE_KEY);
```

Do you guys think it would be useful to add these instructions in the README? I can do a PR with the instructions above if needed.

**Comment by gauravtiwari** [comment]
@jeromedalbert Thanks for sharing 👍  🍰 

Looks good, may be we should have this by default. Feel free to make a PR please to add this

**Comment by jeromedalbert** [comment]
**TLDR:** more complicated and less useful than I thought

I was about to submit a PR with [these code changes](https://github.com/jeromedalbert/webpacker/commit/a6924e60ca89d08b019cc88d9bf3b7bf30c8889b) when I realized something.

It turns out that [Foreman](https://github.com/ddollar/foreman) already loads a `.env` file if it is present. And [Invoker](https://github.com/code-mancers/invoker) loads `.env` and `.env.local` files. So these tools load environment variables from `.env` before starting the Webpack process. When Webpack starts, since the environment variables are already loaded, npm dotenv does not redefine them (this is expected dotenv behavior). So essentially, if you use Foreman or Invoker, you don't need to do anything, it will just silently work with the current Webpacker version.

If you use the code changes above while at the same time using Foreman, it can get convoluted pretty quickly if you have say a `.env` file and a `.env.local` file. Foreman will pick up only the `.env` file, and npm dotenv will not override with variables from `.env.local`, which is confusing and not "ruby dotenv compliant". Same problem if you have a `.env` file and a `.env.development` file.

The code changes are only really useful and fully work as intended only if for some reason you do not use Foreman in development; for example if you manually launch `./bin/webpack-dev-server`. Or if you don't use Ruby at all and have a pure JS app (which defeats the purpose of this issue 😄 ).

Anyways, knowing all of this, it looks like the benefits are less obvious...

**Comment by gauravtiwari** [comment]
@jeromedalbert Oh yes, that makes sense. Sorry confused with another issue where someone tried to run the webpack binary directly and was hitting the same env issue. 

Thanks for detailed explanation and feel free to document the usage whenever you get a chance 👍  I guess useful if someone is running binstubs directly without foreman or invoker right?

**Comment by jeromedalbert** [comment]
Correct. Ok, I'll document this when I get a chance.

**Comment by mike1011** [comment]
> **TLDR:** more complicated and less useful than I thought
> 
> I was about to submit a PR with [these code changes](https://github.com/jeromedalbert/webpacker/commit/a6924e60ca89d08b019cc88d9bf3b7bf30c8889b) when I realized something.
> 
> It turns out that [Foreman](https://github.com/ddollar/foreman) already loads a `.env` file if it is present. And [Invoker](https://github.com/code-mancers/invoker) loads `.env` and `.env.local` files. So these tools load environment variables from `.env` before starting the Webpack process. When Webpack starts, since the environment variables are already loaded, npm dotenv does not redefine them (this is expected dotenv behavior). So essentially, if you use Foreman or Invoker, you don't need to do anything, it will just silently work with the current Webpacker version.
> 
> If you use the code changes above while at the same time using Foreman, it can get convoluted pretty quickly if you have say a `.env` file and a `.env.local` file. Foreman will pick up only the `.env` file, and npm dotenv will not override with variables from `.env.local`, which is confusing and not "ruby dotenv compliant". Same problem if you have a `.env` file and a `.env.development` file.
> 
> The code changes are only really useful and fully work as intended only if for some reason you do not use Foreman in development; for example if you manually launch `./bin/webpack-dev-server`. Or if you don't use Ruby at all and have a pure JS app (which defeats the purpose of this issue smile ).
> 
> Anyways, knowing all of this, it looks like the benefits are less obvious...

The same is explained here as well - https://webpacker-docs.netlify.app/docs/env
